armi/physics/neutronics/tests/test_crossSectionManager.py

```diff
@@ -468,25 +471,57 @@ def test_createRepresentativeBlocksUsingExistingBlocks(self):
         )
         self.assertEqual(origXSIDsFromNew["BA"], "AA")
 
-    def test_interactCoupled_UpdateTrue(self):
-        """ensure that representativeBlocks get populated if timeNode == 0"""
+    def test_interactBOL(self):
+        """Test `BOL` lattice physics update frequency"""
         self.blockList[0].r.p.timeNode = 0
-        self.csm.interactCoupled(iteration=0)
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "BOL"
+        self.csm.interactBOL()
         self.assertTrue(self.csm.representativeBlocks)
 
-    def test_interactCoupled_UpdateFalse(self):
-        """ensure that representativeBlocks remains empty if timeNode == 1"""
+    def test_interactBOC(self):
+        """Test `BOC` lattice physics update frequency"""
+        self.blockList[0].r.p.timeNode = 0
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "BOC"
+        self.csm.interactBOL()
+        self.csm.interactBOC()
+        self.assertTrue(self.csm.representativeBlocks)
+
+    def test_interactEveryNode(self):
+        """Test `everyNode` lattice physics update frequency"""
         self.blockList[0].r.p.timeNode = 1
-        self.csm.interactCoupled(iteration=0)
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "BOC"
+        self.csm.interactBOL()
+        self.csm.interactEveryNode()
         self.assertFalse(self.csm.representativeBlocks)
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "everyNode"
+        self.csm.interactBOL()
+        self.csm.interactEveryNode()
+        self.assertTrue(self.csm.representativeBlocks)
 
-    def test_interactCoupled_Snapshots(self):
-        """ensure that representativeBlocks get populated if runtype is snapshots"""
-        self.csm.cs[CONF_RUN_TYPE] = "Snapshots"
-        self.blockList[0].r.p.timeNode = 2
+    def test_interactFirstCoupled(self):
+        """Test `firstCoupled` lattice physics update frequency"""
+        self.blockList[0].r.p.timeNode = 1
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "everyNode"
+        self.csm.interactBOL()
+        self.csm.interactCoupled(iteration=0)
+        self.assertFalse(self.csm.representativeBlocks)
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "firstCoupled"
+        self.csm.interactBOL()
         self.csm.interactCoupled(iteration=0)
         self.assertTrue(self.csm.representativeBlocks)
 
+    def test_interactAllCoupled(self):
+        """Test `all` lattice physics update frequency"""
+        self.blockList[0].r.p.timeNode = 1
```

Comment:
In the series of tests you added, it doesn't look like you need the `self.blockList[0].r.p.timeNode ...` options for the tests to pass.

Reply:
That's a good point. I was trying to make a test that would have failed with the old code that checked `self.r.p.timeNode`, but once that old code is deleted that piece of the test is effectively obsolete.